lib/modules/@apostrophecms/login/src/apos/components/TheApostropheLogin.vue

```diff
@@ -1,50 +1,107 @@
 <template>
-  <div class="apos-login">
-    <form @submit.prevent="submit">
-      <h2 v-if="error">{{ error }}</h2>
-      <fieldset>
-        <label>Username</label>
-        <input v-model="username" required />
-      </fieldset>
-      <fieldset>
-        <label>Password</label>
-        <input v-model="password" required type="password" />
-      </fieldset>
-      <fieldset>
-        <button type="submit">Log In</button>
-      </fieldset>
-    </form>
+  <div class="apos-login__loader" v-if="contextLoading">Loading<AposSpinner /></div>
+  <div class="apos-login apos-theme-dark" v-else>
+    <div class="apos-login__overlay"></div>
+    <div class="apos-login__menu-overlay"></div>
+    <AposLoginBackground />
+
+    <div class="apos-login__header">
+      <label
+        class="apos-login__project apos-login__project-env"
+        :class="[`apos-login__project-env--${context.env}`]"
+      >
+        {{ context.env }}
+      </label>
+      <label class="apos-login__project apos-login__project-name">{{ context.name }}</label>
+      <label class="apos-login--error">{{ error }}</label>
+    </div>
+
+    <div class="apos-login__body">
+      <form>
+        <AposSchema
```

Comment:
AposSchema is still not following the pattern from ApostropheSchemaEditor where it implements support for v-model and the value it emits as `input` contains both an overall `hasErrors` property and an object with all of the field values. That should be fixed before this goes forward.

Reply:
I see it is fixed in the string field PR, so when that is finalized this PR can be updated, and it should rely on the `hasErrors` property of the modeled value to decide if the form is ready to submit or not.